openml/datasets/functions.py

```diff
@@ -462,14 +469,16 @@ def get_dataset(
     finally:
         if remove_dataset_cache:
             _remove_cache_dir_for_id(DATASETS_CACHE_DIR_NAME, did_cache_dir)
-
-    dataset = _create_dataset_from_description(
-        description, features_file, qualities_file, arff_file, parquet_file, cache_format
-    )
+    if qualities_file:
```

Comment:
mypy gives an arg-type error otherwise, since qualities_file can be None (in exception but mypy doesn't see that). So in run time behavior it is never None so line 472 is always true, since otherwise exception is thrown.